### DISCUSSION TITLE
Next JS and caching (SSG, SSR, cache)

### DISCUSSION THREAD

**Discussion by Jbmanllr** [vercel/next.js#21115] [Help]
Hello, Next JS is an amazing piece of software that i really enjoy working with, however i'm a bit lost on what decision to take with the way to handle caching datas.

My website is more than 200k pages big, and i chose SSR for SEO. So i tested Get Server Side Props and Get Static Props and i was wondering several things:

1 - Using solely get server side props, what would be the best cache solution that would allow me to bust cache ONLY when the content of the page have changed, not just on a timer. Is it possible with next js or only with API reponse caching that monitor PUT, POST on and endpoit?

2 - Using GetStatic Props, is it possible to use incremental static regeneration based on the page changes, and not only on a timer? (like the cache solution i was asking for in the first question)

3 - Is it actually necessary to use a cache solution when using get static props? If yes why exactly, and what would be the best implementation?

Thanks and sorry for my english !

**Comment by FSaldanha** [comment]
Related: #22384

**Comment by stigkj** [comment]
1. Not possible with plain Next.js as of now, as far as I know
2. No, timer is the only supported option
3. When using `getStaticProps`you don't need a cache. The only problem is that all instances of the app must build up its own cache. The easiest way to cache these pages is to put a CDN in front as then the page must only be created by one of the instances and it will be cached "forever".
